src/HealthChecks.MongoDb/MongoDbHealthCheck.cs

```diff
@@ -21,6 +21,12 @@ public MongoDbHealthCheck(string connectionString, string? databaseName = defaul
         }
     }
 
+    public MongoDbHealthCheck(IMongoClient client, string? databaseName = default)
+        : this(client.Settings, databaseName)
+    {
+        _mongoClient[_mongoClientSettings.ToString()] = client;

```

Comment:
We should remove this cache, but this is outside of the scope of this PR. I've created #2148 to track it.

Reply:
Hi @adamsitnik! Since it is not part of the scope of this PR can we close this one?